doc/source/config.rst

```diff
@@ -115,6 +115,21 @@ configuration files, they are merged in reverse order. This means "base"
 configuration paths should be at the end of the list and custom/user paths
 should be at the beginning of the list.
 
+Note that this value must be a list. In Python, this could be set by doing:
+
+.. code-block:: python
+
+    satpy.config.set(config_path=['/path/custom1', '/path/custom2'])
+
+As mentioned above, if setting an environment variable then it must be a
```

Comment:
Thank you, this is already much clearer!
To keep it a bit shorter and avoid repeating the informations, I would propose to delete the "As mentioned above, " here, and rather move the paragraph above to after the code snippets, like:
```
If setting an environment variable then it must be a colon-separated string and must be set **before** calling/importing Satpy. If the environment variable is a single path it will be converted to a list
when Satpy is imported.

.. code-block:: bash
    export SATPY_CONFIG_PATH="/path/custom1:/path/custom2"
    
Satpy will always include the builtin configuration files that it
is distributed with regardless of this setting. When a component supports merging of
configuration files, they are merged in reverse order. This means "base"
configuration paths should be at the end of the list and custom/user paths
should be at the beginning of the list.
```

Reply:
Sounds good and makes sense. I'll edit that after I finish up some other local changes. Thanks.

Reply:
Ok I just pushed a commit with these changes. Do you think the `Note that this value must be a list` part at the bottom is still useful? Is it in a good place?

Reply:
Yes I think it's definitely still useful, I would even move it one or two paragraphs higher (before `If setting an environment variable...`), so that you first show the two examples, and then at the end add the note on the order of the paths (`Satpy will always include...`).

Reply:
Done, but now I'm wondering if the sentence:

> Directories provided will typically include subdirectories for
each component that is being configured (ex. ``readers``, ``writers``, etc).

Is confusing. It sounds like "you should include `/readers` when you specify this, but I'm trying to say the opposite.

Reply:
Probably best clarified with an example, also to illustrate that the `etc/` should be included with `SATPY_CONFIG_PATH`; that may not be immediately obvious to all.

Reply:
Indeed it is a bit confusing. It's also probably good to explicitly give the list of subdirectory names that will be automatically detected and understood by Satpy if contained in one of the paths of `config_path`.

Reply:
Ok I updated it one more time. Let me know what you guys think.

Reply:
Looks good, thank you!